src/hypertable_restrict_info.c

```diff
@@ -524,6 +549,18 @@ gather_restriction_dimension_vectors(const HypertableRestrictInfo *hri)
 														   open->lower_strategy,
 														   open->lower_bound);
 
+				/*
+				 * If we have a nontrivial condition for the second index column
```

Comment:
What does nontrivial mean exactly?

Reply:
Hmm that is weird why do you need backwards scan for constraints on the 2nd index column scan direction shouldnt matter for that.

Reply:
If you have `<=` for the 2nd column, and scan direction is forward, you can't use the 2nd column to find the starting point for the scan. So you use only the first column, and go through the rest of values sequentially. For backwards, you know where the first record is that matches both the conditions on the 1st and the 2nd columns.

More about this in postgres code: https://github.com/postgres/postgres/blob/master/src/backend/access/nbtree/nbtsearch.c#L928

We have this optimization in several other places, so I decided to add it here as well.